Add options `labelFilter` and `managedRecordTypes` in Helm Chart

abaguas opened the pull request:
**Description**

By promoting this options to dedicated values they no longer have to be configured via `extraArgs`.

**Motivation**

In the [K8GB project](https://github.com/k8gb-io/k8gb), a DNS based load balancer, we use external-dns as a chart dependency. We would like to configure all values specific to the controller in the default values of our Chart, and leave to the users the provider configuration. This provider configuration usually includes `extraArgs`. Since `extraArgs` is a list that would be overwritten we would like to keep it empty, otherwise users will have to copy paste our base configuration.

**Checklist**

- [x] Unit tests updated
- [x] End user documentation updated

abaguas commented:
> Thanks for the PR @abaguas, this make sense given the fact that the args are generic.
> 
> Could you please run [`helm-docs`](https://github.com/norwoodj/helm-docs) locally to update the README for the new values.
> 
> Could you also update _charts/external-dns/ci/ci-values.yaml_ to use the new values you've added. Records could just be setting the implicit defaults explicitly and the selector could be a wildcard.

Done, but I have a couple of doubts.

What is the role of the `charts/external-dns/ci/ci-values.yaml` file? It was last updated 1 year ago and I didn't find any code references to it? https://github.com/kubernetes-sigs/external-dns/commits/master/charts/external-dns/ci

What is a wildcard selector? The selector is a string of the form `<key>=<value>`. A `*` would be taken literally which is not what we want. What am I missing?

The README was already updated. Running `helm-docs` resulted in no changes.

stevehipwell commented on charts/external-dns/values.yaml:
```diff
@@ -229,6 +229,12 @@ domainFilters: []
 ## -- Intentionally exclude domains from being managed.
 excludeDomains: []
 
+## -- (string) Filter resources queried for endpoints by label selector
```
```suggestion
# -- (string) Filter resources queried for endpoints by label selector
```

Could you also fix the 2 values above to use a `# -- ` prefix.

abaguas replied:
I see where the mistake came from, thanks. I fix mine and the ones above. The descriptions are now generated

stevehipwell commented on charts/external-dns/values.yaml:
```diff
@@ -229,6 +229,12 @@ domainFilters: []
+## -- (string) Filter resources queried for endpoints by label selector
+labelFilter:
+
+## -- Record types to manage (default: A, AAAA, CNAME)
```
```suggestion
# -- Record types to manage (default: A, AAAA, CNAME)
```

stevehipwell reviewed (changes requested):
> What is the role of the charts/external-dns/ci/ci-values.yaml file? It was last updated 1 year ago and I didn't find any code references to it? https://github.com/kubernetes-sigs/external-dns/commits/master/charts/external-dns/ci

Helm chart testing uses this to set the chart values.

> What is a wildcard selector? The selector is a string of the form <key>=<value>. A * would be taken literally which is not what we want. What am I missing?

You're right. I think for testing purposes it just needs to be a valid selector.

> The README was already updated. Running helm-docs resulted in no changes.

The syntax is incorrect which is why the README descriptions aren't being populated. I've suggested the fixes including the existing ones which look to be what you copied.

abaguas commented:
> > What is the role of the charts/external-dns/ci/ci-values.yaml file? It was last updated 1 year ago and I didn't find any code references to it? https://github.com/kubernetes-sigs/external-dns/commits/master/charts/external-dns/ci
> 
> Helm chart testing uses this to set the chart values.
> 
> > What is a wildcard selector? The selector is a string of the form =. A * would be taken literally which is not what we want. What am I missing?
> 
> You're right. I think for testing purposes it just needs to be a valid selector.
> 
> > The README was already updated. Running helm-docs resulted in no changes.
> 
> The syntax is incorrect which is why the README descriptions aren't being populated. I've suggested the fixes including the existing ones which look to be what you copied.

Thank you for the feedback. I added a valid selector for helm testing and fixed the helm comments 👍

stevehipwell commented:
/ok-to-test

stevehipwell commented:
/lgtm
/approve
